### ISSUE TITLE
1.3.3 tarball doesn't build

### ISSUE THREAD

**Issue by darix** [acassen/keepalived#524]
```
main.c:50:64: error: expected ',' or ';' before 'GIT_DATE'
 #define VERSION_STRING  PACKAGE_NAME " v" PACKAGE_VERSION " (" GIT_DATE ")"
                                                                ^
main.c:57:30: note: in expansion of macro 'VERSION_STRING'
 const char *version_string = VERSION_STRING;  /* keepalived version */
                              ^~~~~~~~~~~~~~
main.c: In function 'parse_cmdline':
main.c:51:47: error: expected ')' before 'GIT_YEAR'
 #define COPYRIGHT_STRING "Copyright(C) 2001-" GIT_YEAR " Alexandre Cassen, <acassen@gmail.com>"
                                               ^
main.c:630:34: note: in expansion of macro 'COPYRIGHT_STRING'
    fprintf(stderr, "\n\n%s\n\n", COPYRIGHT_STRING);
                                  ^~~~~~~~~~~~~~~~
make[2]: *** [Makefile:362: main.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/keepalived-1.3.3/keepalived/core'
make[1]: *** [Makefile:571: all-recursive] Error 1
make[1]: Leaving directory '/home/abuild/rpmbuild/BUILD/keepalived-1.3.3/keepalived'
make: *** [Makefile:367: all-recursive] Error 1
```

**Comment by pqarmitage** [comment]
Many thanks for the report. This has been fixed by commit bd8f6bc and so when the next release of keepalived can be made the issue will be resolved. In the mean time, either pull the latest code from git, or use the attached tarball.
[keepalived-1.3.3.tar.gz](https://github.com/acassen/keepalived/files/779998/keepalived-1.3.3.tar.gz)

**Comment by darix** [comment]
updated opensuse package using this tarball.

**Comment by pqarmitage** [comment]
Reopening this issue so that anyone else who has the same problem can use this.

Will close it again after the next release of keepalived.

**Comment by otheus** [comment]
@pqarmitage The commit bd8f6bc did not fix the problem. I applied the patch (to .in), renegerated the  lib/Makefile, removed the hand-crafted lib/git-commit.h file, and re-ran make with pretty much the same results as @darix . 

The problem has three sources of error. 1. The git-commit.h file is not distributed in the tar file in the official release (in 1.3.3).  2. The git-commit.h is regenerated every make instead of with a "make dist" or via a git commit-hook. A builder who is building from a tar-ball will never have the latest value here.  In essence, it's only for (certain) devs. 3. the make shell script which generates the file doesn't provide default values if for whatever reason git cannot be run or the source is not currently in git (as with a tar ball). 

I submit https://github.com/acassen/keepalived/pull/526

**Comment by pqarmitage** [comment]
@orteus The update appears to have worked for @darix, so I'm not clear what the problem is that you are experiencing.

Regarding the problems you list:
1. The git-commit.h is in the tar file, but it is an empty file. As I explained, that is due to a bug in lib/Makefile.am that is resolved by commit bd8f6bc. I suggest that rather than patching lib/Makefile.in (which is a generated file anyway), you either use the tarball above, or use the latest code from git.

2. git-commit.h is not regenerated every make; it is only regenerated if it would be different from the previous version. Since the tarball should (and now does) include a populated lib/git-commit.h, which has the git version from which the tarball was generated, it does have the latest version. If someone modifies the code that has been extracted from a tarball, well there is nothing we can do about that, but at least the git version on which their code was based will be reported, and there is no other relevant identifier to report.

3. Since the tarball includes git-commit.h, there is no reason to update it or provide default values if building from the tarball.

**Comment by pqarmitage** [comment]
v1.3.4 now released, and builds successfully from tarball.
